doc/sphinx/addendum/type-classes.rst

```diff
@@ -564,6 +573,28 @@ type, like:
 Settings
 ~~~~~~~~
 
+.. _TypeclassesDefaultMode:
+
+.. opt:: Typeclasses Default Mode {| "+" | "-" | "!" }.
+
+   Sets the default mode declaration associated with a :cmd:`Class` or :cmd:`Existing Class`
+   declaration. In Coq <= 8.20, it was set by default to "-", i.e. doing no mode filtering
+   by default. From Coq 8.21 on, it is set to "!", meaning each parameter should be headed by
+   something else than an existential variable for resolution to proceed. If the option is set,
+   then each class declaration uses this default mode for *all* its parameters, unless a
+   :attr:`mode` attribute is used to set the mode explicitly.
+
+   .. _TypeclassesDefaultModeWarning:
```

Comment:
```suggestion
```

Label is not used, can be deleted.  If you want to keep it, you might use the warning name `class-declaration-default-mode` as the label.  Maybe that would be a good convention.  (Or someday we could enhance `.. warn::` to include the warning name in a standard way.)

Reply:
It will be useful in the changelog entry.